Make Jetpack AI Paragraph block editable

romarioraffington opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/Automattic/jetpack/issues/28384

Here's a video explaining proposed changes and possible regressions and improvements: 

https://www.loom.com/share/5993f814ab7b4edfb1b2f23ac004391c

### Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Make the AI Paragraph block editable after AI content is generated


### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Add the AI Paragraph block to post
* Edit the content of the AI content that was generated
* Update or publish the post and verify the content is displayed as expected

### Possible Regression / Improvements

1. ~Typing forward slash `/` in the AI Paragraph block should display the block list. Once a block is chosen a new block should be created below the AI Paragraph block~ This is the typical behavior for blocks  ( except for the Paragraph block )
2. ~Creating an AI Paragraph block after another displays cached content.~ This was a cache issue 

### Does this pull request change what data or activity we track or use?
No

romarioraffington committed:
Made AI Paragraph block editable

Use RichText instead of RawHTML to render text

romarioraffington committed:
Change component "ShowLitteByLittle" to "ShowContent"

ShowContent provides a better description of the component role

romarioraffington committed:
Make "html" the default state for content

romarioraffington committed:
Assign the delay value to a variable

romarioraffington committed:
changelog

romarioraffington committed:
Merge branch 'trunk' into add/made-ai-paragraph-block-editable

n3f commented on projects/plugins/jetpack/extensions/blocks/ai-paragraph/index.js:
```diff
@@ -73,7 +73,7 @@ export const settings = {
 	save: attrs => {
 		const blockProps = useBlockProps.save();
-		return <RawHTML { ...blockProps }>{ attrs.attributes.content }</RawHTML>;
+		return <RichText.Content { ...blockProps } value={ attrs.attributes.content } />;
```
I would wrap this in `p` tags
```suggestion
		return <p {...blockProps}><RichText.Content value={ attrs.attributes.content } /></p>;
```

n3f replied:
This should make the HTML look "better".

n3f reviewed (dismissed):
Works for me.  Great work.

n3f reviewed (commented):
Found an odd bug.  ~~When there is no title~~ the content "flashes".

![Kapture 2023-01-26 at 16 39 00](https://user-images.githubusercontent.com/6354169/214981558-2c31b448-a16a-4fa1-aed9-3e0a844098fa.gif)

Update:

It occurs with a title too.

n3f commented:
Also, not your problem, but this doesn't look correct:

![image](https://user-images.githubusercontent.com/6354169/214981762-99432629-31e6-4c0f-9f51-c4e43d9f1a52.png)

romarioraffington committed:
Update projects/plugins/jetpack/extensions/blocks/ai-paragraph/index.js

Co-authored-by: Brent Nef <nef@a8c.com>

romarioraffington committed:
Set the initial content state to ""

This will prevent the content from flashing when the text is being
progressively written to the canvas

romarioraffington committed:
Fix lint

romarioraffington commented:
> Found an odd bug. ~When there is no title~ the content "flashes".
> 
> ![Kapture 2023-01-26 at 16 39 00](https://user-images.githubusercontent.com/6354169/214981558-2c31b448-a16a-4fa1-aed9-3e0a844098fa.gif)
> 
> Update:
> 
> It occurs with a title too.

Fixed in [7e9a8c8](https://github.com/Automattic/jetpack/pull/28575/commits/7e9a8c87ff0eefceb196f78e6e69a9f76262bc10)

millerf commented on projects/plugins/jetpack/extensions/blocks/ai-paragraph/edit.js:
```diff
@@ -291,12 +295,10 @@ export default function Edit( { attributes, setAttributes, clientId } ) {
-					} }
+				<ShowContent
 					html={ attributes.content }
+					setAttributes={ setAttributes }
```
I think that is a very bad way of doing it, you should rely on a more "React" way and create a "onChange" (and reinstate "onAnimationDone"). This way the component something passed by the parent.

romarioraffington replied:
Fixed in https://github.com/Automattic/jetpack/pull/28575#:~:text=main%20render%20fn-,2b770d6,-romarioraffington%20dismissed%20n3f

millerf reviewed (commented):
LGTM. I would prefer not to have setAttributes passed down to "ShowContent", though...

romarioraffington committed:
Move setAttribute calls to main render fn

romarioraffington commented:
> LGTM. I would prefer not to have setAttributes passed down to "ShowContent", though...

Fixed in https://github.com/Automattic/jetpack/pull/28575#:~:text=main%20render%20fn-,2b770d6,-romarioraffington%20dismissed%20n3f
